source/source_integration_test.go

```diff
@@ -0,0 +1,119 @@
+// limitations under the License.
+
+//go:build integration
+
+package source
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jaswdr/faker"
+	helper "github.com/miquido/conduit-connector-azure-storage/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSource(t *testing.T) {
+	ctx := context.Background()
+	fakerInstance := faker.New()
+
+	var (
+		containerName = "foo"
+
+		cfgRaw = map[string]string{
+			ConfigKeyConnectionString: helper.GetConnectionString(),
+			ConfigKeyContainerName:    containerName,
+			ConfigKeyPollingPeriod:    "1500ms",
+			ConfigKeyMaxResults:       "1",
+		}
+	)
+
+	containerClient := helper.PrepareContainer(t, helper.NewAzureBlobServiceClient(), containerName)
+
+	var (
+		alreadyExistingBlob1Name        = "already-existing-1.txt"
+		alreadyExistingBlob1ContentType = "text/plain; charset=utf-8"
+		alreadyExistingBlob1Contents    = fakerInstance.Lorem().Sentence(16)
+
+		alreadyExistingBlob2Name        = "already-existing-2.txt"
+		alreadyExistingBlob2ContentType = "text/plain; charset=utf-8"
+		alreadyExistingBlob2Contents    = fakerInstance.Lorem().Sentence(16)
+
+		createdWhileWorking1Name        = "created-while-running-1.txt"
+		createdWhileWorking1ContentType = "text/plain; charset=utf-8"
+		createdWhileWorking1Contents    = fakerInstance.Lorem().Sentence(16)
+	)
+
+	require.NoError(t, helper.CreateBlob(containerClient, alreadyExistingBlob1Name, alreadyExistingBlob1ContentType, alreadyExistingBlob1Contents))
+	require.NoError(t, helper.CreateBlob(containerClient, alreadyExistingBlob2Name, alreadyExistingBlob2ContentType, alreadyExistingBlob2Contents))
+
+	time.Sleep(time.Second)
+
+	src := NewSource()
+
+	require.NoError(t, src.Configure(ctx, cfgRaw))
+	require.NoError(t, src.Open(ctx, nil))
+
+	t.Cleanup(func() {
+		_ = src.Teardown(ctx)
+	})
+
+	t.Run("Firstly, snapshot iterator runs and reads all blobs", func(t *testing.T) {
+		// Read the first record from the first page
+		record1, err := src.Read(ctx)
+		require.NoError(t, err)
+		require.True(t, helper.AssertRecordEquals(t, record1, alreadyExistingBlob1Name, alreadyExistingBlob1ContentType, alreadyExistingBlob1Contents))
+		require.NoError(t, src.Ack(ctx, record1.Position))
+
+		// Read the second record from the second page
+		record2, err := src.Read(ctx)
+		require.NoError(t, err)
+		require.True(t, helper.AssertRecordEquals(t, record2, alreadyExistingBlob2Name, alreadyExistingBlob2ContentType, alreadyExistingBlob2Contents))
+		require.NoError(t, src.Ack(ctx, record2.Position))
+
+		// No third record available, backoff
+		// By this time, combined iterator should switch from snapshot to CDC iterator
+		record3, err := src.Read(ctx)
+		require.Equal(t, sdk.Record{}, record3)
+		require.ErrorIs(t, err, sdk.ErrBackoffRetry)
+	})
+
+	t.Run("Secondly, CDC iterator runs and reads all blobs created meanwhile", func(t *testing.T) {
```

Comment:
add more integration tests for edge case scenarios
ex: like invalid config, empty storage, stop the connector after snapshot and start it again ..

Reply:
Updated: added new tests